gateway/constants.ts

```diff
@@ -0,0 +1,68 @@
+export const GATEWAY_URL = 'wss://gateway.discord.gg/?v=8&encoding=json';
+
+// https://discord.com/developers/docs/topics/gateway#identify-identify-connection-properties
+export const GATEWAY_PROPERTIES = {
+  $os: 'linux',
+  $browser: 'DiscordTS',
+  $device: 'DiscordTS',
+
+};
+
+// https://discord.com/developers/docs/topics/opcodes-and-status-codes#gateway-gateway-opcodes
+export enum GATEWAY_OPCODES {
+    DISPATCH = 0,
+    HEARTBEAT = 1,
+    IDENTIFY = 2,
+    PRESENCE_UPDATE = 3,
+    VOICE_STATE_UPDATE = 4,
+    RESUME = 6,
+    RECONNECT = 7,
+    REQUEST_GUILD_MEMBERS = 8,
+    INVALID_SESSION = 9,
+    HELLO = 10,
+    HEARTBEAT_ACK = 11
+}

```

Comment:
5 is `VOICE_SERVER_PING`, 12 is `SYNC_GUILD`, 13 is `SYNC_CALL`

Reply:
I'm lost where is that from

Reply:
From the official Discord API documentation

Reply:
I don't see it in https://discord.com/developers/docs/topics/opcodes-and-status-codes#gateway-gateway-opcodes u59

Reply:
Huh, I always thought they're there but seems not. Anyways those exists — might be jist undocumented.

Reply:
Yep, that's correct, I've checked discordjs and eris, discordjs has voice_ping, while eris has all that you just mentioned.
https://github.com/abalabahaha/eris/blob/master/lib/Constants.js

Reply:
https://github.com/discord/discord-api-docs/blob/ea589a6fdaecf8ace6fa7ddaa871ec7e342083d0/docs/topics/Gateway.md you could cite this

they probably removed it in a later version of the gateway. if you don't want to support an older version of the API than is available, I think there's no pressure for you to add these defs

Reply:
You are actually correct here
5 was removed here https://github.com/discord/discord-api-docs/commit/5ea083cfedeed9a24286299a4ec012e301e16496#diff-2adb841744dbfb715ce95d82e0c7bad2a498da041ee52c1282e2336286b8361a
12 and 13 I believe are undocumented

Reply:
Actually 12 and 13 are both removed, unable to find which commit though

Reply:
We can resolve this correct?

Reply:
Unless we're gonna support backwards compatibility for previous versions of the api then you can

Reply:
Alright I'll resolve then